client/client.go

```diff
@@ -104,18 +102,23 @@ func New(opts ...Option) (*Client, error) {
 	c.closeCh = make(chan struct{})
 	c.workers = make(chan *worker, c.numWorkers)
 	for i := uint(0); i < c.numWorkers; i++ {
-		w := newWorker(c.closeCh, c.httpClient, c.endpoint, c.accessToken, c.maxRetries)
+		w := newWorker(c.pauseForDuration, c.httpClient, c.endpoint, c.accessToken)
```

Comment:
First parameter seems unnecessary.
Side note: we need to enable CI on this repo.